guardrails/async_guard.py

```diff
@@ -48,7 +56,115 @@ class AsyncGuard(Guard):
     the LLM and the validated output stream.
     """
 
-    async def _execute(  # FIXME: Is this override necessary?
+    @classmethod
+    def _from_rail_schema(
+        cls,
+        schema: ProcessedSchema,
+        rail: str,
+        *,
+        num_reasks: Optional[int] = None,
+        tracer: Optional[Tracer] = None,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+    ):
+        guard = super()._from_rail_schema(
+            schema,
+            rail,
+            num_reasks=num_reasks,
+            tracer=tracer,
+            name=name,
+            description=description,
+        )
+        if schema.output_type == OutputTypes.STRING:
+            return cast(AsyncGuard[str], guard)
+        elif schema.output_type == OutputTypes.LIST:
+            return cast(AsyncGuard[List], guard)
+        else:
+            return cast(AsyncGuard[Dict], guard)
+
+    @classmethod
+    def from_pydantic(
+        cls,
+        output_class: ModelOrListOfModels,
+        *,
+        prompt: Optional[str] = None,  # deprecate this too
+        instructions: Optional[str] = None,  # deprecate this too
```

Comment:
Make these comments a TODO so it’s more searchable or can we just deprecate now?

Reply:
These were ad-hoc thoughts I put in there at first but now they are planned for deprecation in 0.5.1 with @dtam's changes.  Will make these TODO's